### PR TITLE
chore: use cdklabs-projen-project-types

### PR DESCRIPTION
**Pull request by mrgrain**
We are standardizing repository setups to improve maintainability.
If you have any questions regarding this change, please reach out to us.

### CODE REVIEW

File: .projenrc.ts [hunk 1]
```diff
@@ -15,8 +17,6 @@ const project = new typescript.TypeScriptProject({
   autoApproveUpgrades: true,
 });
 
-project.package.addField('private', true);
```
Comment (kaizencc) [inline .projenrc.ts:18]: @mrgrain i believe that we default `private: true` in cdklabs projen types, which is why this line is simply deleted right

**Review by kaizencc — COMMENTED** [review]
In addition we need to either 1) click a button in the settings to allow auto merge or 2) set up the automation in cdk-ops to do this for us. @comcalvi may have already started work on a projen task that can configure repository settings.

File: .projenrc.ts (hunk 1 above)
Comment (mrgrain) [reply .projenrc.ts:18]: Yes, that's a manual change a made

File: .npmignore
[the remark this answers is not in the document]
Comment (mrgrain) [reply .npmignore:1]: Guessing that's because it's a private package?

**Comment by mrgrain** [comment]
> In addition we need to either 1) click a button in the settings to allow auto merge or 2) set up the automation in cdk-ops to do this for us. @comcalvi may have already started work on a projen task that can configure repository settings.

I have started clicking these buttons manually. Also need to enable the merge queue via ops me thinks.

File: .npmignore
[the remark this answers is not in the document]
Comment (mrgrain) [reply .npmignore:1]: Yes confirmed, that's why. I think that's okay.

File: .projenrc.ts [hunk 2]
```diff
@@ -1,9 +1,11 @@
-import { typescript } from 'projen';
+import { CdklabsTypeScriptProject } from 'cdklabs-projen-project-types';
 import { DeprecatedTypes } from './projenrc/deprecated-types';
 import { generatePackages, updateReadme } from './projenrc/generate-packages';
 import { UpdateRegistry } from './projenrc/update-registry';
 
-const project = new typescript.TypeScriptProject({
+const project = new CdklabsTypeScriptProject({
+  setNodeEngineVersion: false,
+  stability: 'stable',
```
[the remark this answers is not in the document]
Comment (mrgrain) [reply .projenrc.ts]: Removed.

File: .projenrc.ts [hunk 3]
```diff
@@ -1,22 +1,16 @@
...
+  setNodeEngineVersion: false,
   defaultReleaseBranch: 'main',
   name: 'cdk-cloudformation',
   projenrcTs: true,
   sampleCode: false,
-  autoApproveOptions: {
-    allowedUsernames: ['cdklabs-automation'],
-    secret: 'GITHUB_TOKEN',
-  },
-  autoApproveUpgrades: true,
```
Comment (mrgrain) [inline .projenrc.ts:15]: These are also not needed.

**Pull request merged**